platform/lib/pipeline/markdownDocument.js

```diff
@@ -212,10 +220,30 @@ class MarkdownDocument {
    * @return {Promise}
    */
   save(path) {
+    let content = '';
     const frontmatter = `---\n${yaml.safeDump(this._frontmatter, {'skipInvalid': true})}---\n\n`;
-
     path = path ? path : this._path;
```

Comment:
This line should not be in the middle of the logic that changes the content.
Please put it directly before `writeFile`